docs/GLOSSARY.md

```diff
@@ -1,76 +1,88 @@
 # Glossary
 
-Index of terminology used in Randovania. Certain terms sometimes are used only in the GUI, with more technical ones being used in code or CLI.
+Index of terminology used in Randovania. Some terms are used only in the GUI, with more technical ones being used in code or CLI.
 
 ## Area
 
-TODO
+A small chunk of a game's playable space, often only a single room. May contain one or more [Nodes](#node).
+
+Examples:
+- Phazon Mining Tunnel in Metroid Prime
+- Big Pink in Super Metroid
 
 ## Configuration
 
-TODO
+Has two meanings:
+
+- Colloquially, a term referring to the many types of settings a player may use when generating a game. This includes trick settings, item pool settings, starting items and location, [Gameplay and Cosmetic Patches](#game-patches), and many other things.
+
+- In code, the Configuration class that each game must provide, derived from the BaseConfiguration class. This is where game-specific parts of the [Preset](#preset) are defined.
 
 ## Dangerous Action
 
-TODO
+Any action that the [Generator](#generator) might take which could potentially leave it stranded; a one-way action. See also [Point of No Return](#point-of-no-return).
 
 ## Database
 
-TODO
+The collection of data that represents a game's [Logic](#logic), including the layout of [Worlds](#world) and [Areas](#area), the [Resources](#resource) a player has to progress, the locations of [Pickups](#pickup) and spawn points, and other information.
 
 ## Event
 
-TODO
+A logical [Resource](#resource) representing an event that may be triggered in game. Events can only be completed once, and cannot be undone.
 
 ## Game Patches
 
-TODO
+Has two meainings:
+
+- Colloquially, any modification made to the game. Usually refers to specific changes made to the game, such as cutscene alterations, and not to the randomization process itself.
+
+- In code, refers to the GamePatches class, which holds all per-layout game modifications for a single player. Crucially, this does not include cosmetic patches.
 
 ## Generator
 
-The part of Randovania that takes the [database](#database), the [preset](#preset) settings, a [random number generator](#rng) and decides which [pickups](#pickup) should be placed where, as well as everything else that is randomly decided in the [layout](#layout).
+The part of Randovania that takes the [Database](#database), the [Preset](#preset) settings, a [Random Number Generator](#rng) and decides which [Pickups](#pickup) should be placed where, as well as everything else that is randomly decided in the [Layout](#layout). This is also where [Game Patches](#game-patches) are applied.
 
 ## Hint
 
-TODO
+A piece of text placed in the game world which may give the player information about how [Pickups](#pickup) were placed. This is distinct from [Log Files](#log-file), which contain all details of a generated game, and are generally considered to be a form of cheating in most competitive settings.
```

Comment:
I wouldn't focus on how it's about pickups, because that's not conceptually the case. You could consider the "map station for elevators destination" a hint too.

Reply:
Thanks for the advice, reworked that definition.